shakenfist/tests/test_virt.py

```diff
@@ -171,93 +171,124 @@ def _make_instance(self, mock_get_instance, mock_create_instance):
         )
 
     @mock.patch('shakenfist.db.get_lock')
-    @mock.patch('shakenfist.db.get_instance_attribute',
-                return_value={
-                    'state': 'initial'
-                })
+    @mock.patch('shakenfist.virt.Instance.state',
+                new_callable=mock.PropertyMock)
+    @mock.patch('shakenfist.virt.Instance._db_set_attribute')
     @mock.patch('shakenfist.etcd.put')
-    def test_update_instance_state(self, mock_put, mock_get_attribute,
-                                   mock_lock):
+    def test_update_instance_state(
+            self, mock_put, mock_attribute_set, mock_state_get, mock_lock):
+        mock_state_get.return_value = {
```

Comment:
Would it be more consistent to put the return value in the PropertyMock instantiation?

Reply:
I actually couldn't get that to work... I ended up doing this because its what the docs told me to do.

Reply:
This seems to work. (Ignoring the rubbish spacing)
```
     @mock.patch('shakenfist.db.get_lock')
     @mock.patch('shakenfist.virt.Instance.power_state',
-                new_callable=mock.PropertyMock)
+                new_callable=mock.PropertyMock(return_value={
+                                                'power_state_previous': 'off',
+                                                'power_state': 'on',
+                                                'power_state_updated': 0
+                                            }))
     @mock.patch('shakenfist.virt.Instance._db_set_attribute')
     def test_update_power_state(
             self, mock_attribute_set, mock_power_state_get, mock_lock):
-        mock_power_state_get.return_value = {
-            'power_state_previous': 'off',
-            'power_state': 'on',
-            'power_state_updated': 0
-        }
 
         i = self._make_instance()
         i.update_power_state('off')
```